### DISCUSSION TITLE
Extra SELECTs when inserting with relation {connect:}s

### DISCUSSION THREAD

**Discussion by airhorns** [prisma/prisma#2537] [Q&A] [answered]
Hi folks -- been playing with Prisma in a Redwood JS app and I'm really enjoying the declarative schema modelling! I hooked up some logging to see what kind of SQL queries Prisma is executing underneath and noticed something strange though: when inserting a record and populating it's foreign key field, there's a SELECT statement issued to fetch the related record before actually doing the INSERT. For example, this:

```typescript
await prisma.post.create({
    data: {
      title: "A post with json",
      content: "stuff",
      Blog: { connect: { id: 1 } }
})
```

creates the following queries:

```
prisma:query BEGIN
prisma:query SELECT "public"."Blog"."id" FROM "public"."Blog" WHERE "public"."Blog"."id" = $1 OFFSET $2
prisma:query INSERT INTO "public"."Post" ("blogId","title","metafields","content") VALUES ($1,$2,$3,$4) RETURNING "public"."Post"."id"
prisma:query SELECT "public"."Post"."id", "public"."Post"."blogId", "public"."Post"."title", "public"."Post"."metafields", "public"."Post"."content" FROM "public"."Post" WHERE "public"."Post"."id" = $1 LIMIT $2 OFFSET $3
prisma:query COMMIT
```

Why is the `Blog` record being SELECTed before the `INSERT` happens? If the code knows the ID exactly of the record to connect, couldn't Prisma just insert that value right into the `blogId` column? For write heavy workloads these SELECTs can super kill performance.

Is that on purpose and if so would Prisma be open to making it configurable?

**Comment by Sytten** [comment, marked as answer]
I believe it makes sure that the reference to the record actually exists. In a SQL database with a foreign key that should not be necessary but they plan to support many more types of databases. The devs also said they strive for correctness before performances for now.
I mean for that matter, the last select is also not necessary with a `RETURNING` clause in PostgreSQL at least. I have used prisma for a couple of months now and if performances are what you need now, the tool is not there yet.

**Comment by ludralph** [comment]
Hi @airhorns,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
